File: core/core-api/src/main/java/com/nmh/commerce/controller/ApiControllerAdvice.java
package com.nmh.commerce.controller;

import com.nmh.commerce.support.error.CustomException;
import com.nmh.commerce.support.error.ErrorType;
import com.nmh.commerce.support.response.ApiResponse;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatusCode;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestController;

@Slf4j
@RestController
public class ApiControllerAdvice {

    @ExceptionHandler(CustomException.class)
    public ResponseEntity<ApiResponse<?>> handleCoreException(CustomException e) {
        log.info("CustomException : {}", e.getMessage(), e);
        return new ResponseEntity<>(ApiResponse.error(e.getErrorType(), e.getData()), HttpStatusCode.valueOf(e.getErrorType().getStatus()));
    }

    @ExceptionHandler(Exception.class)
    public ResponseEntity<ApiResponse<?>> handleException(Exception e) {
        log.error("Exception : {}", e.getMessage(), e);
        return new ResponseEntity<>(ApiResponse.error(ErrorType.DEFAULT_ERROR), HttpStatusCode.valueOf(ErrorType.DEFAULT_ERROR.getStatus()));
    }
}

File: core/core-api/src/main/kotlin/com/nmh/commerce/controller/ApiControllerAdvice.kt
package com.nmh.commerce.controller

import com.nmh.commerce.support.error.CustomException
import com.nmh.commerce.support.error.ErrorType
import com.nmh.commerce.support.response.ApiResponse
import org.springframework.http.HttpStatusCode
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestController

@RestController
class ApiControllerAdvice {
    @ExceptionHandler(CustomException::class)
    fun handleCoreException(e: CustomException): ResponseEntity<ApiResponse<*>?> {
        //        ApiControllerAdvice.log.info("CustomException : {}", e.message, e)
        return ResponseEntity<ApiResponse<*>?>(
            ApiResponse.error(e.errorType, e.data),
            HttpStatusCode.valueOf(e.errorType.status),
        )
    }

    @ExceptionHandler(Exception::class)
    fun handleException(e: Exception): ResponseEntity<ApiResponse<*>?> {
        //        ApiControllerAdvice.log.error("Exception : {}", e.message, e)
        return ResponseEntity<ApiResponse<*>?>(
            ApiResponse.error(ErrorType.DEFAULT_ERROR),
            HttpStatusCode.valueOf(
                ErrorType.DEFAULT_ERROR.status,
            ),
        )
    }
}
